File: tests/transpiler/x/fs/dataset_where_filter.fs
// Generated 2025-07-21 15:07 +0700
open System

type Anon1 = {
    mutable name: string
    mutable age: int
}
type Anon2 = {
    mutable name: string
    mutable age: int
}
type Anon3 = {
    mutable name: obj
    mutable age: obj
    mutable is_senior: bool
}
type Anon4 = {
    mutable name: obj
    mutable age: obj
    mutable is_senior: bool
}
let people: Anon2 list = [{ name = "Alice"; age = 30 }; { name = "Bob"; age = 15 }; { name = "Charlie"; age = 65 }; { name = "Diana"; age = 45 }]
let adults: Anon4 list = [ for person in people do if (person.age) >= 18 then yield { name = person.name; age = person.age; is_senior = (person.age) >= 60 } ]
printfn "%s" (string "--- Adults ---")
for person in adults do
printfn "%s" (String.concat " " [string (person.name); string "is"; string (person.age); string (if person.is_senior then " (senior)" else "")])

File: tests/aster/x/pl/dataset_where_filter.pl
:- style_check(-singleton).
:- initialization(main).
main :-
        People = [_{name: 'Alice', age: 30}, _{name: 'Bob', age: 15}, _{name: 'Charlie', age: 65}, _{name: 'Diana', age: 45}],
            Adults = [_{name: 'Alice', age: 30, is_senior: false}, _{name: 'Charlie', age: 65, is_senior: true}, _{name: 'Diana', age: 45, is_senior: false}],
            writeln('--- Adults ---'),
            Person = _{name: 'Alice', age: 30, is_senior: false},
            writeln('Alice is 30 '),
            Person1 = _{name: 'Charlie', age: 65, is_senior: true},
            writeln('Charlie is 65  (senior)'),
            Person2 = _{name: 'Diana', age: 45, is_senior: false},
        writeln('Diana is 45 ').
